s2a.go

```diff
@@ -81,23 +82,27 @@ func NewClientCreds(opts *ClientOptions) (credentials.TransportCredentials, erro
 	if err != nil {
 		return nil, err
 	}
-	return &s2aTransportCreds{
-		info: &credentials.ProtocolInfo{
-			SecurityProtocol: s2aSecurityProtocol,
-		},
-		minTLSVersion: commonpb.TLSVersion_TLS1_3,
-		maxTLSVersion: commonpb.TLSVersion_TLS1_3,
-		tlsCiphersuites: []commonpb.Ciphersuite{
-			commonpb.Ciphersuite_AES_128_GCM_SHA256,
-			commonpb.Ciphersuite_AES_256_GCM_SHA384,
-			commonpb.Ciphersuite_CHACHA20_POLY1305_SHA256,
-		},
-		localIdentity:               localIdentity,
-		targetIdentities:            targetIdentities,
-		isClient:                    true,
-		s2aAddr:                     opts.S2AAddress,
-		ensureProcessSessionTickets: opts.EnsureProcessSessionTickets,
-	}, nil
+	if opts.EnableV2 == true {
```

Comment:
Sorry I just realized this: we should probably just write `if opts.EnableV2 {` instead